cmd/juju/cloud/detectcredentials.go

```diff
@@ -309,18 +387,30 @@ func (c *detectCredentialsCommand) interactiveCredentialsUpdate(ctxt *cmd.Contex
 		}
 		cred := discovered[num-1]
 		// Prompt for the cloud for which to save the credential.
-		cloudName, err := c.promptCloudName(ctxt.Stderr, ctxt.Stdin, cred.defaultCloudName, cred.cloudType)
+		cloudName, err := c.promptCloudName(ctxt.Stderr, ctxt.Stdin, cred.defaultCloudName)
 		if err != nil {
 			fmt.Fprintln(ctxt.Stderr, err.Error())
 			continue
 		}
+		if quit(cloudName) {
+			goto upload
+		}
+
 		if cloudName == "" {
 			fmt.Fprintln(ctxt.Stderr, "No cloud name entered.")
 			continue
 		}
+		cloud, err := common.CloudOrProvider(cloudName, c.cloudByNameFunc)
+		if err != nil {
+			fmt.Fprintln(ctxt.Stderr, err.Error())
+			continue
+		}
+		if cloud.Type != cred.cloudType {
+			fmt.Fprintln(ctxt.Stderr, errors.Errorf("chosen credentials not compatible with a %s cloud", cloud.Type))
```

Comment:
do we want to return error or continue rather than go to next line?

Reply:
Yes, thank you :) good pickup, fixing now :D